pkg/cloudprovider/servicelb.go

```diff
@@ -285,7 +285,10 @@ func (k *k3s) getStatus(svc *core.Service) (*core.LoadBalancerStatus, error) {
 		return nil, err
 	}
 
-	sort.Strings(expectedIPs)
+	expectedIPs, err = filterByIPFamily(expectedIPs, svc)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
With your change to `filterByIPFamily` you can actually get rid of this entirely... the list comes from `k.podIPs` which already calls `filterByIPFamily`.
```suggestion
```